app/deepseg.cc

```diff
@@ -454,18 +458,24 @@ int main(int argc, char* argv[]) try {
 
 		if (raw.rows == 0 || raw.cols == 0) continue; // sanity check
 
-		if (blur_strength) {
-			raw.copyTo(bg);
-			cv::GaussianBlur(bg,bg,cv::Size(blur_strength,blur_strength),0);
-		}
-		ti.prepns = timestamp();
-
 		if (filterActive) {
 			// do background detection magic
 			ai.get_output_mask(mask);
 
+			// get background frame
+			if (pbk) {
+				grab_background(pbk, width, height, bg);
+			} else {
+				if (blur_strength) {
+					raw.copyTo(bg);
+					cv::GaussianBlur(bg,bg,cv::Size(blur_strength,blur_strength),0);
+				}
```

Comment:
Any reason blurring the background is skipped for animated backgrounds? IMHO there's no real reason why this shouldn't be possible.

Reply:
Good point, I think the original logic was that you have a background exactly as you would like it (still or animated), or you wish to blur the original video background and use that. But possibly a poor assumption, we can blur anything :smile: